packages/create-itp-react-app/templates/app/src/core/reducers/ui/features/featuresReducer.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  FEATURES_FETCH_FAILURE,
+  FEATURES_FETCH_REQUEST,
+  FEATURES_FETCH_SUCCESS,
+} from '<PROJECT-NAME>-core/actions/types';
+
+const initialState = {
+  error: null,
+  isFetchingFeatures: false,
+  items: [],
+};
+
+export default (
+  state: any = initialState,
+  { type = '', payload = {} } : { type: string, payload: any},
```

Comment:
In Flow we used to create a convenient [alias](https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVBTAHgBzgJwBcxCBPHDMAFQEEBjQgSzgDsAeKgPjAF4wBvAD5gA1GQoAuMAGdC+RiwDmAGjA4AhqXjqAJgH4pVMIIC+6Oq1lh8vMAAp1DZi0P0mrNrPlLOASl7cju4sANzo+Hb8JOQYUgDkABaMcaoaWnC68UlxYCa+QA) for this:

```js
export type TAction<T> = {| +type: string, payload?: T |}
```

Could be useful here too?

Reply:
using typesafe-actions [ActionType](https://github.com/piotrwitek/typesafe-actions#actiontype) now